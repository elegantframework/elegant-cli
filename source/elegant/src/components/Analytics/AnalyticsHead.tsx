interface Props {
  /**
   * Google Analytics tracking id
   */
  googleAnalyticsID: string
}

/**
 * Create the required Google Analytics head script.
 * @returns A javascript "script" tag for the GA head code.
 */
const AnalyticsHead = ({googleAnalyticsID = ''}: Props) => {
    if(googleAnalyticsID !== '')
    {
        return (
            <script
              dangerouslySetInnerHTML={{
                __html: `
                  window.dataLayer = window.dataLayer || [];
                  function gtag(){dataLayer.push(arguments);}
                  gtag('js', new Date());
                  gtag('config', '${googleAnalyticsID}', {
                    page_path: window.location.pathname,
                  });
                `,
              }}
            />
        );
    }
    else{
        return null;
    }
};

export default AnalyticsHead;