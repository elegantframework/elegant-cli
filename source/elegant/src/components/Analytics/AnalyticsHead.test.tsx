import React from "react";
import renderer from 'react-test-renderer';
import AnalyticsHead from "./AnalyticsHead";

describe("Analytics Head component", () => {
    it('renders an analytics head when an env var is passed in',() => {
        const head = renderer
            .create(<AnalyticsHead googleAnalyticsID="123456"/>)
            .toJSON();
        expect(head).toMatchSnapshot();
    });

    it('renders nothing when no env var is passed in ',() => {
        const head = renderer
            .create(<AnalyticsHead googleAnalyticsID=""/>)
            .toJSON();
        expect(head).toBeNull();
        expect(head).toMatchSnapshot();
    });
});