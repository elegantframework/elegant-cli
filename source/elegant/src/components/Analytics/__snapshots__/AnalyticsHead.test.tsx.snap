// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Head component renders an analytics head when an env var is passed in 1`] = `
<script
  dangerouslySetInnerHTML={
    {
      "__html": "
                  window.dataLayer = window.dataLayer || [];
                  function gtag(){dataLayer.push(arguments);}
                  gtag('js', new Date());
                  gtag('config', '123456', {
                    page_path: window.location.pathname,
                  });
                ",
    }
  }
/>
`;

exports[`Analytics Head component renders nothing when no env var is passed in  1`] = `null`;
